File: 프로그래머스/0/181867. x 사이의 개수/x 사이의 개수.java
class Solution {
    public int[] solution(String myString) {
        int plus = myString.lastIndexOf("x") == myString.length() - 1 ? 1 : 0;
        int[] answer = new int[myString.split("x").length + plus];
        
        int index = 0;
        
        for(int i = 0; i < myString.length(); i++) {
        	int num = myString.charAt(i) == 'x' ? 0 : 1;
        	answer[index] += num;
        	if(num == 0) index++;
        }
        
        
        return answer;
    }
}

File: 프로그래머스/0/181867. x 사이의 개수/x 사이의 개수.c
#include <stdio.h>
#include <stdbool.h>
#include <stdlib.h>

int* solution(const char* myString) {
    int len = 0;
    int idx = 0;
    
    for(int i = 0; myString[i] != '\0'; i++) {
        if(myString[i] == 'x') len++;
    }

    int* answer = (int*)calloc((len + 1), sizeof(int)); 
    
    while(*myString){
        if(*myString == 'x') {
            idx++;
        }else {
            answer[idx]++;
        }
        myString++;
    }
    
    return answer;
}